pkg/controller/direct/sql/normalize.go

```diff
@@ -65,19 +68,29 @@ func NormalizeSQLInstance(ctx context.Context, kube client.Reader, obj *krm.SQLI
 	if err != nil {
 		return nil, err
 	}
+	sourceSQLInstance, err := normalizeSourceSQLInstanceRef(ctx, kube, obj)
```

Comment:
Personally I've been liking converting the refs "in place", I.e. updating them to replace name/namespace refs with external refs.  The nice thing is that then I think we could create a generic function that walks a struct using reflection, finds Refs (by interface) and calls a Resolve function on them (defined on the interface).

But YMMV, I'm watching how this shapes out!  It also doesn't _really_ matter - the bulk of the logic here is in the resolvers anyway.

Reply:
Hmm, that sounds cool! I am hopeful that there will be time to squeeze in some time to use the tools for generating mapping / normalizing functions for SQLInstance in the future. In any case, it's internal (shouldn't affect users), so it will be possible to perform this refactor later.